### PR TITLE
Improve wsgihelpers resilience

### PR DESCRIPTION
**Pull request by MattiSG**

### CODE REVIEW

**Comment by MattiSG** [comment]
Fixes #15.

File: openfisca_web_api/wsgihelpers.py [hunk 1]
```diff
@@ -96,10 +96,13 @@ def respond_json(ctx, data, code = None, headers = None, jsonp = None):
                 for name, value in error.iteritems()
                 if value is not None
                 )
+        if isinstance(error, str):
+            error = data['error'] = dict(
+                message = data['error'],
+                code = 500
```
Comment (cbenz) [inline openfisca_web_api/wsgihelpers.py]: I don't think it is a good idea to hardcode 500, the `code` argument of `respond_json` should be used, without setting a default value for the `code` argument (`code = code or 500`).

But we prefer to be explicit in the caller, so I don't like this part of the patch.
I would prefer to add an assertion, replacing:

```
if isinstance(error, collections.Mapping):
```

by

```
if error is not None:
    assert isinstance(error, collections.Mapping) and error.get('code') is not None
    data = [...etc...]
```

Comment (MattiSG) [inline openfisca_web_api/wsgihelpers.py]: > I don't think it is a good idea to hardcode 500, the code argument of respond_json should be used
That is the case  ;)
I'm just setting a default for `error['code']`, which is evaluated lines 111-112 as `code = code or error['code']`.

> I would prefer to add an assertion
If you have a failing assertion in this part of the code, you will crash and send a text-only `Internal Server Error` response.

This patch is not great, but it is what amounted to the least refactor of the existing heuristics in the `if error:` block.

**Comment by MattiSG** [comment]
Can I provide you with any additional information?

File: openfisca_web_api/wsgihelpers.py (hunk 1 above)
Comment (cbenz) [inline openfisca_web_api/wsgihelpers.py]: Still not convinced but hard to write, please come IRL :)

**Pull request closed without merging**